modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/hooks/useThunk.es.js

```diff
@@ -12,20 +12,31 @@
  * details.
  */
 
-import {PagesVisitor} from '../../util/visitors.es';
+import {useIsMounted} from 'frontend-js-react-web';
+import {useRef} from 'react';
 
-export default (pages, properties) => {
-	const {fieldInstance} = properties;
-	const pageVisitor = new PagesVisitor(pages);
+export const useThunkDispatch = (dispatch) => {
+	const isMounted = useIsMounted();
 
-	return Promise.resolve(
-		pageVisitor.mapFields((field) => {
-			const focused = field.name === fieldInstance.name;
+	const thunkDispatch = useRef((action) => {
```

Comment:
This is very similar (identical?) to what we have [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/core/hooks/useThunk.js); we should probably move that into `frontend-js-react-web` soon.

Created https://issues.liferay.com/browse/LPS-116631 for that.

Reply:
Yeah, I think it would be great to have there. It is a great way to deal with side effects.